python/pdo/common/key_value_swig/swig_utils.cpp

```diff
@@ -0,0 +1,103 @@
+/* Copyright 2018 Intel Corporation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <stdlib.h>
+#include <stdio.h>
+#include <string>
+
+#include <Python.h>
+
+#include <iostream>
+
+#include "log.h"
+#include "swig_utils.h"
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+static PyObject* glogger = NULL;
+void SetLogger(
+    PyObject* inLogger
+    )
+{
+    if (glogger) {
+        Py_DECREF(glogger);
+    }
+    glogger = inLogger;
+    if (glogger) {
+        Py_INCREF(glogger);
+    }
+} // _SetLogger
+
+// XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+static PyGILState_STATE gstate;
+static void PyLog(
```

Comment:
I don't see that logging is used in the cpp files (or perhaps it is. though elsewhere in the common/crypto libs), but... it can be useful.